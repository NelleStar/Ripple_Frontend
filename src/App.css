body {
  background-color: #D8E3E0;
}
