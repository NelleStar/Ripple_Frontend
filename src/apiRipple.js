import axios from "axios";

const BASE_URL = process.env.REACT_APP_BASE_URL || "http://localhost:3001";

class RippleApi {
  // the token for interaction with Ripple database will be stored here.
  static token;

  static async request(endpoint, data = {}, method = "get") {
    console.debug("DB Call:", endpoint, data, method);

    // pass token in header
    const url = `${BASE_URL}/${endpoint}`;
    const headers = { Authorization: `Bearer ${RippleApi.token}` };
    const params = method === "get" ? data : {};

    try {
      return (await axios({ url, method, data, params, headers })).data;
    } catch (err) {
      console.error("API Error:", err.response);
      let message = err.response.data.error.message;
      throw Array.isArray(message) ? message : [message];
    }
  }

  // login a user and get token
  static async loginUser(data) {
    try {
      let res = await this.request(`auth/token`, { ...data }, "post");
      return res;
    } catch (err) {
      return err;
    }
  }

  // register a user and get token
  static async registerUser(data) {
    try {
      let res = await this.request(`auth/register`, { ...data }, "post");
      return res;
    } catch (err) {
      return err;
    }
  }

  // get all users back from db
  static async getUsers() {
    try {
      const res = await this.request(`users/`);
      return res;
    } catch (err) {
      return err;
    }
  }

  // get single user back based on username
  static async getUser(username) {
    try {
      let res = await this.request(`users/${username}`);
      return res;
    } catch (err) {
      return err;
    }
  }

  // patch single users profile
  static async patchUser(username, data) {
    try {
      let res = await this.request(`users/${username}`, { ...data }, "patch");
      return res;
    } catch (err) {
      return err;
    }
  }

  // get all waves in db
  static async getWaves() {
    try {
      let res = await this.request(`waves/`);
      return res;
    } catch (err) {
      return err;
    }
  }

  // get single wave based on id
  static async getWave(id) {
    try {
      let res = await this.request(`waves/${id}`);
      return res;
    } catch (err) {
      return err;
    }
  }

  // create a new wave
  static async postWave(data) {
    try {
      let res = await this.request(`waves/`, { ...data }, "post");
      return res;
    } catch (err) {
      return err;
    }
  }

  // patch wave based on id and data
  static async patchWave(id, data) {
    try {
      let res = await this.response(`waves/${id}`, { ...data }, "patch");
      return res;
    } catch (err) {
      return err;
    }
  }

  // Method for deleting a single wave
  static async deleteWave(id) {
    try {
      let res = await this.request(`waves/${id}`, {}, "delete");
      return res;
    } catch (err) {
      return err;
    }
  }

  // Methods for handling comments
  static async getComments(waveId) {
    try {
      let res = await this.request(`waves/${waveId}/comments`);
      return res;
    } catch (err) {
      return err;
    }
  }

  //get a comment
  static async getComment(waveId, commentId) {
    try {
      let res = await this.request(`waves/${waveId}/comments/${commentId}`);
      return res;
    } catch (err) {
      return err;
    }
  }

  //post a new comment
  static async postComment(waveId, data) {
    try {
      let res = await this.request(
        `waves/${waveId}/comments`, { ...data }, "post"
      );
      return res;
    } catch (err) {
      return err;
    }
  }

  //patch a comment
  static async patchComment(waveId, commentId, data) {
    try {
      let res = await this.request(
        `waves/${waveId}/comments/${commentId}`,
        { ...data },
        "patch"
      );
      return res;
    } catch (err) {
      return err;
    }
  }

  //delete a comment
  static async deleteComment(waveId, commentId) {
    try {
      let res = await this.request(
        `waves/${waveId}/comments/${commentId}`,
        {},
        "delete"
      );
      return res;
    } catch (err) {
      return err;
    }
  }
}

export default RippleApi;

