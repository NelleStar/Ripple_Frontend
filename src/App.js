import { useState, useEffect } from "react";
import { BrowserRouter, Routes, Route, Navigate } from "react-router-dom";

import userContext from "./userContext";
import NavBar from "./NavBar/NavBar";
import Home from "./Components/Home/Home";
import UsersList from "./Components/Users/UsersList";
import UserDetails from "./Components/Users/UserDetails";
import WavesList from "./Components/Waves/WavesList";

import LoginForm from "./Forms/LoginForm/LoginForm";
import SignUpForm from "./Forms/SignUpForm/SignUpForm";
import UserForm from "./Forms/UserForm/UserForm";

import RippleApi from "./apiRipple";

import "./App.css";
import NewCommentForm from "./Forms/CommentForms/NewCommentForm";
import WaveCard from "./Components/Waves/WaveCard";

// ======================================================================== //

function App() {
  // set state
  const [user, setUser] = useState({});

  // store data for user in state
  const getUser = async (username) => {
    const res = await RippleApi.getUser(username);
    console.log(`App getUser results:`, res);
    setUser({ ...res.user, token: RippleApi.token });
  };

  // collect username and token
  useEffect(() => {
    let username = localStorage.getItem("username");
    let token = localStorage.getItem("token");
    console.log(`App useEffect Username and Token:`, username, token);

    // if both are there update them
    if (username && token) {
      RippleApi.token = token;
      getUser(username);
    } else {
      RippleApi.token = "";
    }
  }, []);

  //update local storage, token and state
  const logIn = (data) => {
    localStorage.setItem("username", data.username);
    localStorage.setItem("token", data.token);
    RippleApi.token = data.token;
    getUser(data.username);
    console.log(`app.js logIn`)
  };

  // log user out
  const logOut = (data) => {
    localStorage.clear();
    RippleApi.token = "";
    setUser({});
    console.log(`app.js logOut`)
  };

  return (
    <div className="App">
      <userContext.Provider value={user}>
        <BrowserRouter>
          <NavBar logOut={logOut} />
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/login" element={<LoginForm logIn={logIn} />} />
              <Route path="/signup" element={<SignUpForm logIn={logIn} />} />
              <Route path="/users" element={<UsersList />} />
              <Route path="/users/:username" element={<UserDetails />} />
              <Route
                path="/users/:username/edit"
                element={<UserForm getUser={getUser} />}
              />
              <Route path="/waves" element={<WavesList />} />
              <Route path="*" element={<Navigate to="/"/>}/>            
          </Routes>
        </BrowserRouter>
      </userContext.Provider>
    </div>
  );
}

export default App;
