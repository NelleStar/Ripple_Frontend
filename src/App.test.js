import { render, fireEvent, waitFor } from '@testing-library/react';
import App from './App';
import { apiRipple } from './apiRipple';
import { BrowserRouter } from 'react-router-dom';

it ('renders without crashing', () => {
  render(<App />);
});

it('matches Snapshot', () => {
  const {asFragment} = render(<App />);
  expect(asFragment).toMatchSnapshot()
});

it('renders login link', () => {
  const {getByText} = render(<App />)
  expect(getByText('login')).toBeInTheDocument()
});