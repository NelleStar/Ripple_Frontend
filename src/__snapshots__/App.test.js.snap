// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Snapshot 1`] = `[Function]`;
